src/server/search/doc_index.cc

```diff
@@ -46,6 +48,19 @@ void TraverseAllMatching(const DocIndex& index, const OpArgs& op_args, F&& f) {
 
 }  // namespace
 
+search::FtVector BytesToFtVector(string_view value) {
+  DCHECK_EQ(value.size() % sizeof(float), 0u);
+  search::FtVector out(value.size() / sizeof(float));
+
+  // Create copy for aligned access
+  unique_ptr<float[]> float_ptr = make_unique<float[]>(out.size());
+  memcpy(float_ptr.get(), value.data(), value.size());
```

Comment:
Does anyone know a simpler approach?

Reply:
If you're casting `value` from a byte-view to a float-view, could you instead cast it into a `Span<float>` instead of copying it (or to a `float[]`?)
I guess that would be simpler, and will also avoid a copy

Reply:
The issue I'm trying to solve is that the byte view can be misaligned as a float view (a float is 4 bytes, so is requires: ptr % 4 = 0)

The byte view is just part of a string, so I can't really change its alignment.

Reply:
Now you `DCHECK()` that string is aligned (line 52). Do you plan to support a case in which it's not aligned? If so, how?

Reply:
I check _size_ in line 52. Pointer address alignment is something else

Reply:
Oh, gotcha! Sorry.
Do you want to have an aligned access as an optimization, i.e. to get it to run faster?
If so, perhaps you could have a pointer pointing to the string_view initially, but if it points to an unaligned string, you'll copy it like you do and have the pointer point there.
Am I even close to understanding what you're trying to solve?

Reply:
> aligned access as an optimization

I read it once, so I don't care about it as an optimization. The issue is correctness

At first I forgot about alignment, but got a runtime error in my debug build (unaligned read). That's when I added the defensive copy. I always thought unaligned reads are better avoided at all and can cause UB.

Reply:
Misaligned memory access is an area of the language that I'm not familiar with. From what I read, yeah, it seems to be UB to even get to a point where the memory is unaligned :shrug: 
Sorry, I don't have any better ideas than what you currently have (which seems to be marginally UB as is, but it's probably OK from a practical standpoint)

Reply:
Why are we not just memcpying the whole array into `out`? This should be legal.

Reply:
I agree, that could save a copy